packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/MethodCallHandlerImpl.java

```diff
@@ -75,19 +76,24 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
     String action = convertAction((String) call.argument("action"));
     Integer flags = call.argument("flags");
     String category = call.argument("category");
-    Uri data = call.argument("data") != null ? Uri.parse((String) call.argument("data")) : null;
-    Bundle arguments = convertArguments((Map<String, ?>) call.argument("arguments"));
+    String stringData = call.argument("data");
+    Uri data = call.argument("data") != null ? Uri.parse(stringData) : null;
+    Map<String, ?> stringMap = call.argument("arguments");
+    Bundle arguments = convertArguments(stringMap);
     String packageName = call.argument("package");
+    String component = call.argument("componentName");
     ComponentName componentName =
-        (!TextUtils.isEmpty(packageName)
-                && !TextUtils.isEmpty((String) call.argument("componentName")))
-            ? new ComponentName(packageName, (String) call.argument("componentName"))
-            : null;
+            (packageName != null
+                    && component != null
+                    && !TextUtils.isEmpty(packageName)
+                    && !TextUtils.isEmpty(component))
+                    ? new ComponentName(packageName, component)
+                    : null;
```

Comment:
nit: I would find this easier to read as a regular declaration followed by an if statement. This seems like a lot of logic for a ternary.